eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -177,21 +177,26 @@ stages:
                         parameters:
                           WorkingDirectory: $(Pipeline.Workspace)/vcpkg
 
-                      # SkipCheckingForChanges is true to skip the commit step
-                      # (which is already done by Update-VcpkgPort.ps1)
                       - template: /eng/common/pipelines/templates/steps/create-pull-request.yml
                         parameters:
+                          PushWorkingDirectoryChanges: false
                           RepoOwner: $(VcpkgPrRepoOwner)
                           RepoName: $(VcpkgPrRepoName)
                           WorkingDirectory: $(Pipeline.Workspace)/vcpkg
                           PrBranchName: $(PrBranchName)
                           PRTitle: $(PrTitle)
                           PRBody: Update vcpkg ports for Azure SDK release. This release may contain multiple ports.
-                          SkipCheckingForChanges: true
                           BaseBranchName: $(DefaultBranch)
                           OpenAsDraft: ${{ parameters.TestPipeline }}
                           PRLabels: "auto-merge"
                           CloseAfterOpenForTesting: '${{ parameters.TestPipeline }}'
+                          # If PublishToVcpkg is true, this template blindly attempts to create a PR. In that case
+                          # changes must already be checked into the branch for this template to succeed.
+                          # If PublishToVcpkg is false, the non-skipped checking for changes will result in no attempt
+                          # to open a PR because there will be no detected changes.
+                          # Use dictionary syntax because the template inserts the literal string into a `condition:`
+                          # element.
+                          SkipCheckingForChanges: variables['PublishToVcpkg']
```

Comment:
Why wouldn't `$(PublishToVcpkg)` not work? By the time the runtime conditions evaluate that should have the value true or false already which would avoid needing the change in the common template.

Reply:
OK I think I can see part of the problem you are trying to solve here which is that the parameter is used in both a runtime condition as well as a PS inline expression and in those cases, you access the variable in different ways. That can be a little tricky.

Reply:
Aside from pulling the template apart and re-writing it to handle this scenario specifically in the C++ repo, this seems like the best way to reuse code and extend this to support runtime evaluation of whether a package should go to vcpkg (a thing that can only be evaluated at runtime). 

We could wrap the vcpkg publish process up in a separate job and have a vcpkg betas publish job determine wither vcpkg publish should run at all.

Reply:
Couldn't we control this by making sure we set HasChanges=false in the case you don't want a PR? That is essentially what is happening in the template but you are blindly setting HasChanges=true when PublishToVcpkg=true but if you set it to false when that is false I think you will also get what you are after.

Reply:
I think you're right. I'll give it a shot.

Reply:
You were right.